packages/gitmoji-changelog-markdown/src/index.js

```diff
@@ -1,17 +1,57 @@
 const fs = require('fs')
 const path = require('path')
 const handlebars = require('handlebars')
+const { isEmpty } = require('lodash')
 
 const MARKDOWN_TEMPLATE = path.join(__dirname, 'template.md')
 
-function convert(changelog) {
+function convert({ meta, changes }) {
   const template = fs.readFileSync(MARKDOWN_TEMPLATE, 'utf-8')
 
   const generateMarkdown = handlebars.compile(template)
 
+  const changelog = changes.map(version => ({
+    ...version,
+    groups: version.groups.map(group => ({
+      ...group,
+      commits: group.commits.map(commit => ({
+        ...commit,
+        hash: getShortHash(commit.hash, meta.repoInfo),
+        subject: autolink(commit.subject, meta.repoInfo),
+        message: autolink(commit.message, meta.repoInfo),
+        body: autolink(commit.body, meta.repoInfo),
+      })),
+    })),
+  }))
```

Comment:
Can we improve this with immutadot ?

Reply:
Something like that should work 😄 
```js
update(changes, 'groups[:].commits[:]', commit => ({
  ...commit,
  hash: getShortHash(commit.hash, meta.repository),
  subject: autolink(commit.subject, meta.repository),
  message: autolink(commit.message, meta.repository),
  body: autolink(commit.body, meta.repository),
}))
```

Reply:
I should not use `map` instead ?

```js
map(changes, 'groups[:].commits[:]', commit => ({
  ...commit,
  hash: getShortHash(commit.hash, meta.repository),
  subject: autolink(commit.subject, meta.repository),
  message: autolink(commit.message, meta.repository),
  body: autolink(commit.body, meta.repository),
}))
```

Reply:
It works with `update`, but the path is : `'[:].groups[:].commits[:]'`